themes/src/main/resources/theme/base/login/resources/js/authChecker.js

```diff
@@ -12,6 +14,13 @@ addEventListener("beforeunload", () => {
   }
 });
 
+setTimeout(() => {
+  const kcAuthSessionHash = getKcAuthSessionHash();
+  if (kcAuthSessionHash !== initialKcAuthSessionHash) {
+    location.reload();
```

Comment:
I wonder that `location.reload();` is possible issue IMO when thinking about this? It assumes that the HTTP request from client application can be easily repeated by reloading the page. I think this works fine for SAML and OIDC GET requests (so maybe 90% of scenarios). There are still remaining 10% of cases like: SAML Post-binding requests, OIDC POST requests (The OAuth/OIDC as well as `AuthorizationEndpoint` supports POST as well even if it is not commonly used), OAuth PAR (pushed authorization request) and maybe more. It seems to me that for those `location.reload()` won't work well :-(


Not 100% sure how to improve... I can think of something like introducing new endpoint (probably on `LoginActionsService`) for "merging" authentication sessions. Maybe something like this:

- Cookie `KC_AUTH_SESSION_HASH` will need to be changed to use encryption (likely by realm AES key) instead of hashing (so probably cookie renamed again)

- If the change is detected, the browser will send the HTTP request to the endpoint, which will have parameters like:
- `initialAuthSessionId` (the original value of hashed authentication session ID)
- `tabId` (should be accessible to this tab)
- `newAuthSessionId` (maybe not needed as it can be read from the cookie in that endpoint, but again not 100% sure if some concurrency in browser can cause the different value than read by the cookie due the many browser tabs loading at the same time and overwriting the cookies etc)

Endpoint can read the AuthenticationSessionModel from existing `intialAuthSessionId` and `tabId` and "move" it to new RootAuthenticationSession and re-render the page.

Can something like this work? Or any easier way to still achieve this working for things like POST requests, PAR etc without introducing another endpoint?

Reply:
you're right that this scenario while rare can happen. I had already experimented with this solution by adding the hash in the `AuthenticationSessionBean`, reading it on the web page, and comparing it with the cookie after the timeout. 

The only thing I didn’t like was hashing the auth session id every time the web page is loaded because I wasn't sure how to store it, since a new auth session is created and the root auth session doesn't have any auth notes. Would be fine hashing the auth session id every time on the web page? Or should we find a way to temporarily store it?

Reply:
I've tested the reload of OIDC POST requests both manually and with `authChecker.js`, and it works. The issue with POST requests is that you might see the "confirm form resubmission" message when reloading the page. In Chrome, this message does not appear when calling `location.reload();` but it does appear in Firefox. @jonkoops, do you have any suggestions?
OAuth PAR would be left out since it requires a different mechanism. The scenario you described could be implemented as a follow-up. WDYT?

Reply:
Yes, it works for me to address it as a follow-up rather than introducing another endpoint

Reply:
There is no way to work around the form submission confirmation mechanism unfortunately. It is meant to provide additional security for the user so they do not accidentally perform an action that might be destructive. As you found out browser vendors have different interpretations of how this should work.